### PR TITLE
Add tags to force-style unit tests

### PR DESCRIPTION
**Pull request by rbberger**
**Summary**

Adds an optional "tags" entry in the force style test YAML. This is a comma-separated list of keywords, which are parsed by CMake and added as labels for CTest.  This allows more fine-grained filtering of tests. Any newly-generated
YAML file automatically adds the "generated" tag.

CTest Examples:
```bash
# exclude tests with 'unstable' label
ctest -LE unstable

# only execute test with specified label
ctest -L slow
```

**Related Issue(s)**

#2978
#2986

**Author(s)**

@rbberger 

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

**Implementation Notes**

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

**Comment by rbberger** [comment]
@akohlmey so far I've only added `slow` and `unstable`. Feel free to change those. It would help if you could add the necessary tags for tests where you already know about problems.

File: unittest/force-styles/CMakeLists.txt [hunk 1]
```diff
@@ -73,24 +80,30 @@ if(FFT_SINGLE)
 endif()
 foreach(TEST ${MOL_PAIR_TESTS})
   string(REGEX REPLACE "^.*mol-pair-(.*)\.yaml" "MolPairStyle:\\1" TNAME ${TEST})
+  extract_tags(TEST_TAGS ${TEST})
   add_test(NAME ${TNAME} COMMAND test_pair_style ${TEST} WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
   set_tests_properties(${TNAME} PROPERTIES ENVIRONMENT "LAMMPS_POTENTIALS=${LAMMPS_POTENTIALS_DIR};PYTHONPATH=${TEST_INPUT_FOLDER}:$ENV{PYTHONPATH}")
+  set_tests_properties(${TNAME} PROPERTIES LABELS "${TEST_TAGS}")
```
Comment (junghans) [inline unittest/force-styles/CMakeLists.txt:86]: Doesn't `TEST_TAGS` have to be a cmake list instead of a space separated string?

**Comment by akohlmey** [comment]
@rbberger added more "slow" tags to get an about 1:1 split with respect to time and 60:430 split for tests.

File: unittest/force-styles/CMakeLists.txt (hunk 1 above)
Comment (rbberger) [reply unittest/force-styles/CMakeLists.txt:86]: `TEST_TAGS` is a CMake list.

Comment (junghans) [reply unittest/force-styles/CMakeLists.txt:86]: Oh nice, today I learned `string(REGEX MATCHALL ...)` returns a list, great!

**Comment by junghans** [comment]
Did you take the `unstable` tag from the tests failing in #2978?

**Review by junghans — APPROVED** [review]

**Comment by akohlmey** [comment]
@rbberger this has some updates for epsilon values and unstable tags added where it seems justified.

**Review by akohlmey — APPROVED** [review]

**Review by stanmoore1 — APPROVED** [review]

**Pull request merged**